tests/unit/structure/pbc.test.ts

```
@@ -342,3 +354,647 @@ test(`pbc_dist optimization advanced scenarios`, () => {
+    expect(image_atoms.length).toBeGreaterThanOrEqual(expected_min_images)
+    expect(image_atoms.length).toBeLessThanOrEqual(expected_max_images)
+
+    // Test that all image atoms have valid positions
+    for (const [original_idx, image_xyz] of image_atoms) {
+      // Original index should be valid
+      expect(original_idx).toBeGreaterThanOrEqual(0)
+      expect(original_idx).toBeLessThan(structure.sites.length)
+
+      // Image position should be finite and valid
+      expect(image_xyz).toHaveLength(3)
+      expect(image_xyz.every((coord) => isFinite(coord))).toBe(true)
```

Biome (1.9.4)

[error] isFinite is unsafe. It attempts a type coercion. Use Number.isFinite instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isFinite instead.


(lint/suspicious/noGlobalIsFinite)